frontend/src/components/maps/BasemapToggle.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { useState } from 'react';
+import './BasemapToggle.scss';
+
+export type BaseLayer = {
+  name: string;
+  url: string;
+  attribution: string;
+  thumbnail: string;
+};
+
+export type BasemapToggleEvent = {
+  current: BaseLayer;
+  previous: BaseLayer;
+};
+
+type BasemapToggleProps = {
+  baseLayers: BaseLayer[];
+  onToggle?: (e: BasemapToggleEvent) => void;
+};
+
+const BasemapToggle: React.FC<BasemapToggleProps> = props => {
+  // TODO: fade out when map is panning/zooming
+  const [updating] = useState(false);
```

Comment:
As per the **TODO** comment this to signal a future enhancement where the map widgets would kind of fade out when you drag/zoom around the map. This is not part of this story and as such I left it out but wanted to put the initial placeholder code for that